jaworski_cellseg/label_counting_widget.py

```diff
@@ -1,9 +1,53 @@
 from magicgui import magicgui
 import napari
 from skimage.measure import regionprops
+import pandas as pd
 
 
-def create_label_counting_widget(viewer: napari.Viewer):
+def generate_statistics(regions, physical_sizes, save_file_path):
+    # Initialize a list to hold the properties data
+    data = []
+
+    # Loop through each region and filter for the labels of interest
+    for region in regions:
+        # Extract properties and add them to the data list as a dictionary
+        region_data = {
+            "label": region.label,
+            "area": region.area,
+            "centroid_row": region.centroid[2],
+            "centroid_col": region.centroid[1],
+            "centroid_depth": region.centroid[0],
+            "bbox": region.bbox,  # Add more properties as needed
+        }
+
+        min_slice, min_row, min_col, max_slice, max_row, max_col = region.bbox
+
+        # Calculate the center pixel of the bounding box
+        center_pixel = (
+            (min_col + max_col) // 2,
+            (min_row + max_row) // 2,
+            (min_slice + max_slice) // 2,
+        )
```

Comment:
You can create these columns after creating the data frame. Vectorized operations are much faster than doing this in a loop.

Reply:
I vectorized the outer loop and modularized the logic into helper functions. Let me know if that works too. I think is more readable.